relayer/processor/types.go

```diff
@@ -175,11 +248,14 @@ func (c ConnectionStateCache) Merge(other ConnectionStateCache) {
 	}
 }
 
-// Clone makes a copy of the ConnectionStateCache so it can be used by other threads.
-func (c ConnectionStateCache) Clone() ConnectionStateCache {
-	n := make(ConnectionStateCache, len(c))
+// FilterForClient makes a filtered copy of the ConnectionStateCache
+// for a single client ID so it can be used by other threads.
```

Comment:
Go code typically, and in this case specifically, is concerned about goroutine safety rather than thread safety.

Reply:
updated comment